mmv1/products/apphub/Workload.yaml

```diff
@@ -0,0 +1,230 @@
+  result: !ruby/object:Api::OpAsync::Result
+    path: response
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: done
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: error
+    message: message
+update_verb: :PATCH
+update_mask: true
+parameters:
+  - !ruby/object:Api::Type::String
+    name: location
+    description: 'Part of `parent`.  Full resource name of a parent Application. Example: projects/{HOST_PROJECT_ID}/locations/{LOCATION}/applications/{APPLICATION_ID} '
+    url_param_only: true
+    required: true
+    immutable: true
+  - !ruby/object:Api::Type::String
+    name: applicationId
+    description: 'Part of `parent`.  Full resource name of a parent Application. Example: projects/{HOST_PROJECT_ID}/locations/{LOCATION}/applications/{APPLICATION_ID}'
+    url_param_only: true
+    required: true
+    immutable: true
+  - !ruby/object:Api::Type::String
+    name: workloadId
+    description: 'The Workload identifier. '
+    url_param_only: true
+    required: true
+    immutable: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: name
+    output: true
+    description: "Identifier. The resource name of the Workload. Format:\"projects/{host-project-id}/locations/{location}/applications/{application-id}/workloads/{workload-id}\" "
+  - !ruby/object:Api::Type::String
+    name: displayName
+    description: 'User-defined name for the Workload. '
+  - !ruby/object:Api::Type::String
+    name: description
+    description: 'User-defined description of a Workload. '
+  - !ruby/object:Api::Type::NestedObject
+    properties:
+      - !ruby/object:Api::Type::String
+        name: uri
+        description: 'Output only. The underlying compute resource uri. '
+        output: true
+    output: true
+    name: workloadReference
+    description: 'Reference of an underlying compute resource represented by the Workload. '
+  - !ruby/object:Api::Type::NestedObject
+    properties:
+      - !ruby/object:Api::Type::String
+        name: gcpProject
+        description: "Output only. The service project identifier that the underlying
+          cloud resource resides in. Empty for non cloud resources. "
+        output: true
+      - !ruby/object:Api::Type::String
+        name: location
+        description: "Output only. The location that the underlying compute resource resides
+          in (e.g us-west1). "
+        output: true
+      - !ruby/object:Api::Type::String
+        name: zone
+        description: "Output only. The location that the underlying compute resource resides
+          in if it is zonal (e.g us-west1-a). "
+        output: true
+    output: true
+    name: workloadProperties
+    description: 'Properties of an underlying compute resource represented by the Workload. '
+  - !ruby/object:Api::Type::String
+    name: discoveredWorkload
+    diff_suppress_func: 'DiscoveredResourceProjectNumberDiffSuppress'
```

Comment:
I believe this can just use `diff_suppress_func: tpgresource.ProjectNumberDiffSuppress`. Did you try that? Then you could remove the custom code file.

Reply:
We checked that out. But our format doesn't match with that function. Please refer to https://github.com/GoogleCloudPlatform/magic-modules/pull/10132#discussion_r1517505047